### PR TITLE
Add userCourse type

### LINKED ISSUE #2: Add UserCourse, Course, and Assignment Types
### Feature request
------------------------
Add UserCourse, Course, and Assignment Type definitions

### PR DESCRIPTION
**Pull request by jessehartloff**
## Proposed changes
--------------------
Added the UserCourse type to the shared types.

## Types of changes
--------------------
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist
--------------------
- [x] My changeset covers only what is described above (no extraneous changes)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged downstream

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 export * from './types/auth.types'
 export * from './types/user.types'
+export * from './types/user-course.type'
```
Comment (NicholasMy) [inline src/index.ts]: We probably want to stick with the `.types` extension for consistency.

File: src/types/user-course.type.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export type UserCourse = {
+  id?: number
+  userId: number
+  courseId: number
+  dropped?: boolean
+  level?: string
```
Comment (MikeDrewitt) [inline src/types/user-course.type.ts]: If this can be only certain strings then this type should only be 

`'instructor' | 'ta' | 'student'`

Comment (jessehartloff) [reply src/types/user-course.type.ts]: You can do that?? TpyeScript ftw!

**Comment by jessehartloff** [comment]
Discussion in related PR https://github.com/UBAutograding/devU-api/pull/18#discussion_r662389485

File: src/types/user-course.types.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+}
+export const userCourseRoles: UserCourseRole[] = [
+  { label: 'Student', value: 'student' },
+  { label: 'TA', value: 'ta' },
+  { label: 'Instructor', value: 'instructor' },
+]
+
+// I was unable to find a way to remove the duplication here. I want to create a type based on the
+// userCourseRoles.value to avoid having each value in 2 places. If anyone knows how to do this, please share!
+export const userCourseLevels = ['student', 'ta', 'instructor'] as const
```
Comment (MikeDrewitt) [inline src/types/user-course.types.ts]:
```suggestion
export const userCourseLevels = userCourseRoles.map(u => u.value) as const
```

**Review by MikeDrewitt — CHANGES_REQUESTED** [review]

File: src/types/user-course.types.ts (hunk 3 above)
Comment (jessehartloff) [reply src/types/user-course.types.ts]: I get this error when I try that
`TS1355: A 'const' assertions can only be applied to references to enum members, or string, number, boolean, array, or object literals.`

**Review by MikeDrewitt — APPROVED** [review]

**Pull request merged**